tests/sub_group/sub_group_api.cpp

```diff
@@ -170,6 +217,18 @@ TEST_CASE("sub-group api", "[sub_group]") {
   // if possible, reduce size by one to attempt to create incomplete sub-groups
   if (local_size_dim > 1) {
     --local_size_dim;
+    // if required size for result buffer for current work group dimension size
+    // exceed maximum memory allocation size on device reduce initial
+    // local_size_dim (i.e. current local_size_dim + 1) to nearest multiple two
```

Comment:
```suggestion
    // If required size for result buffer for current work group dimension size
    // exceeds maximum memory allocation size on device, reduce initial
    // local_size_dim (i.e. current local_size_dim + 1) to nearest multiple of two
```
But it is unclear what we get at the end. A multiple of a power-of-2?